pkg/services/devicegroups.go

```
@@ -141,17 +144,38 @@ func (s *DeviceGroupsService) CreateDeviceGroup(deviceGroup *models.DeviceGroup)
 }
 
 // GetDeviceGroupByID gets the device group by ID from the database
-func (s *DeviceGroupsService) GetDeviceGroupByID(ID string) (*models.DeviceGroup, error) {
-	var deviceGroup models.DeviceGroup
+func (s *DeviceGroupsService) GetDeviceGroupByID(ID string) (*models.DeviceGroupDetails, error) {
+	//var deviceGroup models.DeviceGroup
+	var deviceGroupDetails models.DeviceGroupDetails
 	account, err := common.GetAccountFromContext(s.ctx)
 	if err != nil {
-		return nil, new(AccountNotSet)
+		fmt.Printf("\n Error account %v \n", err)
```

you should return the error related to not get the account